src/utils/anonymousUserEventManager.ts

```diff
@@ -0,0 +1,356 @@
+            this.updateCart(event);
+            break;
+          }
+          case UPDATE_USER: {
+            this.updateUser({ dataFields: event });
+            break;
+          }
+          default:
+            break;
+        }
+
+        localStorage.removeItem(SHARED_PREFS_ANON_SESSIONS);
+        localStorage.removeItem(SHARED_PREFS_EVENT_LIST_KEY);
+      }
+    }
+  }
+
+  private async storeEventListToLocalStorage(
+    newDataObject: any,
+    shouldOverWrite: boolean
+  ) {
+    const strTrackEventList = localStorage.getItem(SHARED_PREFS_EVENT_LIST_KEY);
+    let previousDataArray = [];
+
+    if (strTrackEventList) {
+      previousDataArray = JSON.parse(strTrackEventList);
+    }
+
+    if (shouldOverWrite) {
+      const trackingType = newDataObject[SHARED_PREFS_EVENT_TYPE];
+      const indexToUpdate = previousDataArray.findIndex(
+        (obj: any) => obj[SHARED_PREFS_EVENT_TYPE] === trackingType
+      );
+      if (indexToUpdate !== -1) {
+        const dataToUpdate = previousDataArray[indexToUpdate];
+
+        previousDataArray[indexToUpdate] = {
+          ...dataToUpdate,
+          ...newDataObject
+        };
+      } else {
+        previousDataArray.push(newDataObject);
+      }
+    } else {
+      previousDataArray.push(newDataObject);
+    }
+
+    localStorage.setItem(
+      SHARED_PREFS_EVENT_LIST_KEY,
+      JSON.stringify(previousDataArray)
+    );
+    const criteriaId = await this.checkCriteriaCompletion();
+    if (criteriaId !== null) {
+      this.createKnownUser(criteriaId);
+    }
+  }
+
+  private getCurrentTime = () => {
+    return new Date().getTime();
+  };
+
+  private getWebPushOptnIn(): string {
+    const notificationManager = window.Notification;
+    if (notificationManager && notificationManager.permission === 'granted') {
+      return window.location.hostname;
+    } else {
+      return '';
+    }
+  }
+
+  track = (payload: InAppTrackRequestParams) => {
+    return baseIterableRequest<IterableResponse>({
+      method: 'POST',
+      url: '/events/track',
+      data: payload,
+      validation: {
+        data: trackSchema
+      }
+    });
+  };
+
+  updateCart = (payload: UpdateCartRequestParams) => {
+    return baseIterableRequest<IterableResponse>({
+      method: 'POST',
+      url: '/commerce/updateCart',
+      data: {
+        ...payload,
+        user: {
+          ...payload.user,
+          preferUserId: true
+        }
+      },
+      validation: {
+        data: updateCartSchema
+      }
+    });
+  };
+
+  trackPurchase = (payload: TrackPurchaseRequestParams) => {
+    return baseIterableRequest<IterableResponse>({
+      method: 'POST',
+      url: '/commerce/trackPurchase',
+      data: {
+        ...payload,
+        user: {
+          ...payload.user,
+          preferUserId: true
+        }
+      },
+      validation: {
+        data: trackPurchaseSchema
+      }
+    });
+  };
+
+  updateUser = (payload: UpdateUserParams = {}) => {
```

Comment:
we shouldn't call this if there's no payload

Reply:
This method has been removed from anonymousUserEventManager.ts